packages/jest-haste-map/src/index.js

```diff
@@ -820,8 +820,17 @@ class HasteMap extends EventEmitter {
     }
 
     if (this._whitelist) {
-      const match = filePath.match(this._whitelist);
-      return !match || match.length > 1;
+      const whitelist = this._whitelist;
+      const match = whitelist.exec(filePath);
+      const matchEndIndex = whitelist.lastIndex;
```

Comment:
The regexp always has the `g` flag in this case, we control that here: https://github.com/ide/jest/blob/a77e2d255af4fbd7f6057664c3311aed3255b092/packages/jest-haste-map/src/index.js#L107

We could use `match[0].length`, I think both approaches are good enough.